KinConnections/static/js/search.js

```diff
@@ -0,0 +1,75 @@
+const connectorsList = document.getElementById('connectorsList');
+const searchBar = document.getElementById('connectorsSearchBar');
+let connectors = [];
+
+searchBar.addEventListener('keyup', (e) => {
+    const searchString = e.target.value.toLowerCase().trim();
+
+    if (searchString.length === 0) {
+        displayConnectors(connectors);
+        return;
+    }
+
+    const filteredConnectors = connectors.filter((connector) => {
+        return (
+            connector.first_name.toLowerCase().includes(searchString) ||
+            connector.last_name.toLowerCase().includes(searchString) ||
+            connector.education.toLowerCase().includes(searchString) ||
+            connector.bio.toLowerCase().includes(searchString) ||
+            connector.location.toLowerCase().includes(searchString) ||
+            connector.title.toLowerCase().includes(searchString) ||
+            connector.region_current.toLowerCase().includes(searchString)
+        );
+    });
+    displayConnectors(filteredConnectors);
+});
+
+const loadConnectors = () => {
+    connectorsCookieValue = getAndCleanCookie("connectors");
+    connectors = JSON.parse(connectorsCookieValue);
+    displayConnectors(connectors);
+};
+
+const getAndCleanCookie = (name) => {
+    const cookies = document.cookie.split(';');
+    for (let i = 0; i < cookies.length; i++) {
+        let c = cookies[i].trim().split("=")
+        if (c[0] === name) {
+            c = c.slice(1, c.length + 1).join("=").split("'");
+            c[0] = c[0].replace('\"', '');
+            c[c.length - 1] = c[c.length - 1].replace('\"', '');
+            c = c.join('"');
+            c = c.replace(RegExp(/\\054/g), ",").replace(/True/g, '0');
+            return c
+        }
+    }
+    return "";
+};
+
+const displayConnectors = (connectors) => {
+    const htmlString = connectors
+        .map((connector) => {
+            return `
+            <li>
+                <div class="col-lg-4 col-md-6 portfolio-item">
```

Comment:
Also had to remove the filtering options on the `jinja` code that was being used to filter based on categories. We might need to rethink that implementation or maybe I just need to morph this to accommodate for that. I haven't looked into the specifics of that yet. Something that still needs some work before this PR is ready to go.